src/Workspaces/Core/Portable/Workspace/Solution/ProjectInfo.cs

```diff
@@ -449,6 +465,7 @@ internal sealed class ProjectAttributes : IChecksummedObject, IObjectWritable
                 Optional<string?> filePath = default,
                 Optional<string?> outputPath = default,
                 Optional<string?> outputRefPath = default,
+                Optional<CompilationOutputFilePaths> compilationOutputPaths = default,
```

Comment:
Do we only use the Optional if it's a reference type, more or less? (VersionStamp is different.) I suspect we might not have a constraint on Optional only because it sometimes got used as a hack around not having nullable reference types, but we might have been stricter for public API.

Reply:
There are existing `Optional<bool>` parameters. I don't think it matters if it's `Optional<T>` or `Nullable<T>`. I'm using Optional for consistency. Eventually, we'll just use records for this.